app/assets/javascripts/submission_obfuscation.js

```diff
@@ -0,0 +1,70 @@
+(function() {
+    let SubmissionObfuscation = {};
+
+    // Retrieve the "obfuscate" parameter from the URL
+    let urlParams = new URLSearchParams(window.location.search);
+    window.shouldNotObfuscate = (urlParams.get('obfuscate') === 'false');
+
+    // Store the original student2 name
+    SubmissionObfuscation.originalStudent2Name = null;
+
+    SubmissionObfuscation.maskStudentName = function() {
+        // console.log("JS Script here onwards");
+        // console.log(window.shouldNotObfuscate)
+    
+        // Check if the global flag for obfuscation is false; if so, return without obfuscating
+        if (window.shouldNotObfuscate) {
+            // console.log("Names are not obfuscated")
+            return;
+        }
+    
+        // console.log("Masking student name now");
+    
+        // Masking inside all h4 elements to get the student pairings
+        let h4Elements = document.querySelectorAll('h4');
```

Comment:
hmm is it possible select by a dedicated class instead of h4
this will allow the search to be more targeted

Reply:
likewise for other query selector functions

Reply:
Yeap, I will rectify